helix-core/src/syntax.rs

```diff
@@ -116,6 +120,8 @@ pub struct LanguageServerConfiguration {
     #[serde(default)]
     #[serde(skip_serializing_if = "Vec::is_empty")]
     pub args: Vec<String>,
+    #[serde(default = "default_timeout")]
+    pub timeout: u64,
```

Comment:
I think many stuff here are not documented in the book, can you please put it there?